app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionController.scala

```diff
@@ -31,17 +31,18 @@ import uk.gov.hmrc.play.http.HeaderCarrier
 import scala.concurrent.Future
 
 @Singleton
-class SubscriptionController @Inject() (override val messagesApi: MessagesApi) (implicit appConfig: AppConfig)
+class SubscriptionController @Inject() (override val messagesApi: MessagesApi, override val authConnector: AuthConnector) (implicit appConfig: AppConfig)
   extends FrontendController with I18nSupport with Actions {
 
-  override protected def authConnector: AuthConnector = FrontendAuthConnector
-
-  val showCheckAgencyStatus: Action[AnyContent] = AuthorisedFor(NoOpRegime, GGConfidence).async { implicit authContext: AuthContext => implicit request =>
-    ensureAffinityGroupIsAgent {
-      Ok(html.subscribe())
-    }
+  val showCheckAgencyStatusBody: (AuthContext) => (Request[AnyContent]) => Future[Result] = {
```

Comment:
The reason for breaking out `showCheckAgencyStatusBody` like this is to make unit testing easier.

`showCheckAgencyStatusBody` can be tested without auth mocking to make the `AuthorisedFor` check pass.

This also avoids the need for the Play app to be started in `SubscriptionControllerSpec` (which is fairly slow). This is because for the `AuthorisedFor` check to pass we'd need to add `.withSession(sessionKeysForMockAuth: _*)` to the `FakeRequest` and `.withSession()` needs a running app because it uses Play's `Crypto` object to sign the session cookie and in turn `Crypto` requires a running app.